proxy/src/test/java/com/wavefront/agent/listeners/otlp/OtlpGrpcMetricsHandlerTest.java

```diff
@@ -807,4 +841,100 @@ public void exponentialCumulativeHistogram() {
 
     EasyMock.verify(mockReportPointHandler);
   }
+
+  @Test
+  public void appRelatedResAttrsShouldBeIncluded() {
+    String applicationKey = "application";
+    String serviceKey = "service.name";
+    String shardKey = "shard";
+    String clusterKey = "cluster";
+
+    EasyMock.reset(mockReportPointHandler);
+
+    Map<String, String> annotations =
+        ImmutableMap.<String, String>builder()
+            .put(applicationKey, "some-app-name")
+            .put("service", "some-service-name")
+            .put(shardKey, "some-shard-name")
+            .put(clusterKey, "some-cluster-name")
+            .build();
+
+    ReportPoint wfMetric =
+        OtlpTestHelpers.wfReportPointGenerator().setAnnotations(annotations).build();
+    mockReportPointHandler.report(wfMetric);
+    EasyMock.expectLastCall();
+
+    EasyMock.replay(mockReportPointHandler);
+
+    NumberDataPoint otelPoint = NumberDataPoint.newBuilder().build();
+    Metric otelMetric = OtlpTestHelpers.otlpGaugeGenerator(otelPoint).build();
+    Resource resource =
+        Resource.newBuilder()
+            .addAttributes(OtlpTestHelpers.attribute(applicationKey, "some-app-name"))
+            .addAttributes(OtlpTestHelpers.attribute(serviceKey, "some-service-name"))
+            .addAttributes(OtlpTestHelpers.attribute(shardKey, "some-shard-name"))
+            .addAttributes(OtlpTestHelpers.attribute(clusterKey, "some-cluster-name"))
+            .build();
+    ResourceMetrics resourceMetrics =
+        ResourceMetrics.newBuilder()
+            .setResource(resource)
+            .addScopeMetrics(ScopeMetrics.newBuilder().addMetrics(otelMetric).build())
+            .build();
+    ExportMetricsServiceRequest request =
+        ExportMetricsServiceRequest.newBuilder().addResourceMetrics(resourceMetrics).build();
+    subject.export(request, emptyStreamObserver);
+
+    EasyMock.verify(mockReportPointHandler);
+  }
+
+  @Test
+  public void appRelatedResAttrsCanBeExcluded() {
+    boolean shouldIncludeOtlpAppTagsOnMetrics = false;
+
+    subject =
+        new OtlpGrpcMetricsHandler(
+            mockReportPointHandler,
+            mockHistogramHandler,
+            preprocessorSupplier,
+            DEFAULT_SOURCE,
+            false,
+            shouldIncludeOtlpAppTagsOnMetrics);
+    String applicationKey = "application";
+    String serviceKey = "service.name";
+    String shardKey = "shard";
+    String clusterKey = "cluster";
+
+    EasyMock.reset(mockReportPointHandler);
+
+    ReportPoint wfMetric = OtlpTestHelpers.wfReportPointGenerator().build();
+    assertFalse(wfMetric.getAnnotations().containsKey(applicationKey));
+    assertFalse(wfMetric.getAnnotations().containsKey(serviceKey));
```

Comment:
Shouldn't this be asserting that the WF metric doesn't contain a tag named `service`, rather than `service.name`?

Reply:
No, this specific test case is to add code coverage for changing the default configuration `otlpAppTagsOnMetricsIncluded`. With this test case, we are not expecting any tags on a metric.